docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+Configuration Annotations
+=========================
+
+The 1.12.x version of Minecraft Forge introduced novel mod configuration system based on annotations. This new system relies on Forge ASM as to automatically discover mod configuration classes and their respective configurable fields as to make working with configurations as easy as possible. Ideally this new system will encourage the standard implementation of mod configurations as to afford end-users with maximally configurable modded Minecraft experiences.
+
+
+Basics
+-------------------------
+
+The annotation-based configuration system revolves around the `@Config` annotation, found in the `net.minecraftforge.common.config` package of Minecraft Forge. The `@Config` annotation is intended to be applied to any class that contains user-configurable fields, requiring only the id of the mod that owns the configuration for basic usage. Following the application of the `@Config` annotation to a configuration class, Forge will then resolve and manage any `public static` fields in the class as configuration elements.
+
+!!! important
+
+    There is currently no way to have Forge ignore a `public static` field in a `@Config` class, so make sure not to place any non-user configurable `public static`
+
+Consequently, the simplest possible usage of the annotation-based configuration system would be as follows.
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  public static boolean myBooleanSetting = false;
+
+}
+````
+
+The prior usage would then create a boolean property `general.myBooleanSetting` with a default value of `false` in a configuration file named `my_mod_id.conf` which is managed by Forge.
+
+Whereas the prior example used only the Java primitive `boolean`, the annotation-based configuration system will accept all Java primitives, and their respective boxed forms. Furthermore, the annotation system is additionally capable of accepting `String` and `enum` values.
+
+!! important
+
+    By default the annotation-based configuration system loads all mod configurations exactly once, directly after the mod is constructed but before the mod enters pre-initialization.
+
+
+Comments
+-------------------------
+
+Often, despite our best efforts in naming, the property key alone does not do justice in describing the function of a given configuration property. In this case, or moreso in general it would be nice as to be able to give the end-user more information as to what a configuration element actually does. For this reason, the `@Comment` annotation was created.
```

Comment:
This has a lot of unnecessary fluff. I would reword it as

>The `@Comment` annotation can be used to add additional information to configuration properties that might not fit in the property key alone.

Reply:
The goal was to stress the motivation behind the `@Comment` annotation, given that understanding the reasoning behind a given element can provide great insight into the proper usage of the element.